pennylane/devices/default_qubit.py

```diff
@@ -479,6 +480,22 @@ def expval(self, observable, shot_range=None, bin_size=None):
 
             return np.real(ev.toarray()[0])
 
+        if observable.name == "Hamiltonian":
```

Comment:
@josh146  this is the addition to default.qubit... `sparse_ham` is basically a dictionary of entries, and the entries should be taken from PauliX.matrix etc...